### PR TITLE
expect: add recipe for v5.45.4

### PR DESCRIPTION
**Pull request by nehaljwani**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/expect/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+autoconf
+( cd testsuite
+  autoconf -I.. )
+
+./configure --prefix=$PREFIX --build=$BUILD --host=$HOST --with-tclinclude=$PREFIX/include
+make -j ${CPU_COUNT}
+make test
+make -j ${CPU_COUNT} install
+
+mv $PREFIX/lib/tcl*/expect${PKG_VERSION}/libexpect${PKG_VERSION}.so $PREFIX/lib
```
Comment (scopatz) [inline recipes/expect/build.sh:11]: Does `make install` not do this for some reason? And are you sure that this move doesn't mess up any RPATHs?

File: recipes/expect/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    - autoconf
+    - automake
+    - make
+  host:
+    - tk
+
+test:
+  commands:
+    - expect -v
+    - echo 'package require Expect' | tclsh
```
Comment (scopatz) [inline recipes/expect/meta.yaml:66]: Is this a meaningful test?

File: recipes/expect/build.sh (hunk 1 above)
Comment (nehaljwani) [reply recipes/expect/build.sh:11]: It's the side effect of a patch, and no it doesn't mess up RPATHs, it's actually a fix for a mess :)
xref: https://src.fedoraproject.org/rpms/expect/blob/f33/f/expect.spec#_123

File: recipes/expect/meta.yaml (hunk 2 above)
Comment (nehaljwani) [reply recipes/expect/meta.yaml:66]: The expect package provides a tcl package which can be imported, which the command 'unbuffer' (also provided by this package) relies on (but can't be tested in the test section)
```
$ head /usr/bin/unbuffer 
#!/bin/sh
# -*- tcl -*-
# The next line is executed by /bin/sh, but not tcl \
exec tclsh8.6 "$0" ${1+"$@"}

package require Expect
```

**Comment by scopatz** [comment]
This should probably have a run_exports

**Comment by nehaljwani** [comment]
> This should probably have a run_exports

Added.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**